File: backend/src/main/java/pt/isel/rugby/model/AthletePractice.java
package pt.isel.rugby.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "athletePractice")
@Data
public class AthletePractice implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private Boolean physio;

    @Column
    private Boolean regular;

    @ManyToOne
    @JsonIgnore
    private Practice practice;

    @ManyToOne
    private Athlete athlete;
}

File: frontend2/Rugby/src/app/classes/associations/AthletePractice.ts
import {Practice} from '../practice';
import {Athlete} from '../athlete';

export class AthletePractice {
  constructor(
    private id?: number,
    public athlete?: Athlete,
    public physio?: boolean,
    public regular?: boolean
  ) {
    this.id = id ? id : 0;
    this.athlete = athlete ? athlete : new Athlete();
    this.physio = physio ? physio : false;
    this.regular = regular ? regular : false;
  }
}
